openmc/source.py

```diff
@@ -377,7 +400,10 @@ def to_tuple(self):
                 self.delayed_group, self.surf_id, self.particle.value)
 
 
-def write_source_file(source_particles, filename, **kwargs):
+def write_source_file(
+    source_particles: 'typing.Iterable[openmc.SourceParticle]',
```

Comment:
Simple solution on this one:
```suggestion
    source_particles: typing.Iterable[SourceParticle],
```